tx_json.go

```diff
@@ -0,0 +1,457 @@
+	if err != nil {                                                                 // Check for errors
+		return nil, err // Return found error
+	}
+
+	code, err := parseContractCode(parsedData) // Parse contract code
+	if err != nil {                            // Check for errors
+		return nil, err // Return found error
+	}
+
+	_, err = payload.Write(gasLimit[:]) // Write to buffer
+	if err != nil {                     // Check for errors
+		return nil, err // Return found error
+	}
+
+	_, err = payload.Write(functionPayloadLength[:4]) // Write to buffer
+	if err != nil {                                   // Check for errors
+		return nil, err // Return found error
+	}
+
+	_, err = payload.Write(functionPayload) // Write to buffer
+	if err != nil {                         // Check for errors
+		return nil, err // Return found error
+	}
+
+	_, err = payload.Write(code) // Write contract code to buffer
+	if err != nil {              // Check for errors
+		return nil, err // Return found error
+	}
+
+	return payload.Bytes(), nil // Return payload
+}
+
+// parseBatch parses a transaction payload with the batch tag.
+func (parser *Parser) parseBatch(data []byte) ([]byte, error) {
+	payload := bytes.NewBuffer(nil) // Initialize buffer
+
+	parsedData, err := parseJSONBytes(data) // Parse data
+	if err != nil {                         // Check for errors
+		return nil, err // Return found error
+	}
+
+	if !parsedData.Exists("payloads") { // Check no value
+		return nil, ErrNilField // Return nil field error
+	}
+
+	transactions := parsedData.GetArray("payloads") // Get payloads
+
+	var batchLength [8]byte // Initialize batch length buffer
+
+	binary.LittleEndian.PutUint64(batchLength[:], uint64(len(transactions))) // Write batch length to buffer
+
+	payload.Write(batchLength[:4]) // Write batch length
+
+	for _, transaction := range transactions { // Iterate through transactions
+		json, err := transaction.StringBytes() // Get JSON
+		if err != nil {                        // Check for errors
+			return nil, err // Return found error
+		}
+
+		txParser := NewParser(string(transaction.GetStringBytes("tag"))) // Initialize parser
+
+		txPayload, err := txParser.ParseJSON(json) // Parse JSON
+		if err != nil {                            // Check for errors
+			return nil, err // Return found error
+		}
+
+		_, err = payload.Write(txPayload) // Write payload
+		if err != nil {                   // Check for errors
+			return nil, err // Return found error
+		}
+	}
+
+	return payload.Bytes(), nil // Return payload
+}
+
+/*
+	END TAG HANDLERS
+*/
+
+/*
+	BEGIN PARSER HELPER METHODS
+*/
+
+// parseOperation gets the operation of a particular stake transaction.
+func parseOperation(json *fastjson.Value) (byte, error) {
+	if !json.Exists("operation") { // Check no value
+		return byte(0), ErrNilField // Return nil field error
+	}
+
+	switch string(json.GetStringBytes("operation")) { // Handle different operations
+	case "0x00":
```

Comment:
0x00, 0x01, and 0x02 as strings can be simplified to a JSON number 0, 1, 2, 3, 4, ... etc. Would be good to validate that the operation code is less than `sys.TagBatch` as well, and that the number is not negative.